### PR TITLE
Add ActionBar.

### PR DESCRIPTION
**Pull request by na9da**
### What this PR does
Required for https://github.com/TerriaJS/nsw-digital-twin/issues/567

- A new floating menu at the bottom of the map that looks like:
  image
  This will be used by the SceneEditor, but other plugins can also make use of it.
- Refactors some of the Portal code in Terria

### Test me

The `ActionBar` is currently not used by any feature in core Terria, so it cannot be tested from the CI. 

### Checklist

- [ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
- [ ] I've updated relevant documentation in `doc/`.
- [ ] I've updated CHANGES.md with what I changed.
- [ ] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

File: lib/Styled/Button.tsx [hunk 1]
```diff
@@ -188,13 +188,21 @@ export const Button: React.FC<ButtonProps> = (props) => {
...
+          <IconSpan
             css={iconProps && iconProps.css}
+            margin={
+              // Apply left or right margin only when the button content is not empty
+              props.children
+                ? props.rightIcon
+                  ? "margin-left: 8px"
+                  : "margin-right: 8px"
+                : null
+            }
```
Comment (na9da) [inline lib/Styled/Button.tsx:200]: Previously, a margin used to be shown even when the contents where empty, this bloated the buttons that only had an Icon and no text. This fixes it.

File: lib/ReactViews/ActionBar/ActionBarPortal.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  left: 0;
+  right: 0;
+  margin: auto;
+
+  /* Animate slide in-out */
+  transition: all 0.2s;
+  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
+`;
+
+export default withViewState(ActionBarPortal);
```
Comment (tephenavies) [inline lib/ReactViews/ActionBar/ActionBarPortal.tsx]: Prefer use of `useViewState` and reduce passing viewState through props to reduce noise. Could you update StandardUserInterface/Portal to use `useViewState`? Then I think you won't need viewState here.

File: lib/ReactViews/Workflow/WorkflowPanelPortal.tsx [hunk 3]
```diff
@@ -37,4 +34,4 @@ const Container = styled.div<{ show: boolean }>`
   opacity: ${(p) => (p.show ? 1 : 0)};
 `;
 
-export default withViewState(WorkflowPanelContainer);
+export default withViewState(WorkflowPanelPortal);
```
Comment (tephenavies) [inline lib/ReactViews/Workflow/WorkflowPanelPortal.tsx]: I think this `withViewState` also can be removed when Portal uses useViewState.

**Review by tephenavies — CHANGES_REQUESTED** [review]

**Comment by na9da** [comment]
Thanks for the feedback @steve9164! I have updated the code to replace explicit `viewState` props with `useViewState()` hook.

**Review by tephenavies — APPROVED** [review]
Thanks @na9da!

**Comment by tephenavies** [comment]
Actually, should there be a changelog entry in this PR, or is it included in a different PR?

**Comment by na9da** [comment]
Thanks @steve9164! 

> Actually, should there be a changelog entry in this PR, or is it included in a different PR?

I have updated the changelog in the latest commit. Kept it for last to avoid frequent merge conflicts.

**Pull request merged**